docs/04-api-reference/front-commerce-core/ServerEvent.mdx

```diff
@@ -0,0 +1,45 @@
+---
+title: "ServerEvent"
+sidebar_position: 8
+description:
+  "The ServerEvent class define ServerEvent format used in Front-Commerce to be
+  published to the server event worker."
+---
+
+<p>{frontMatter.description}</p>
+
+## Usage:
+
+```ts
+import { ServerEvent } from "@front-commerce/core/events";
+
+// We define a static event Name for our event
+const MyEventName = "MyEvent";
+
+// Definition of the payload
+type MyEventPayload = {
+  sku: string;
+  price: boolean;
+};
+
+export default class MyEvent extends ServerEvent<MyEventPayload> {
+  static event_type = MyEventName;
```

Comment:
**issue:** I don't think (nor know why) this should be declared as a `static`. Scalar values are not complex enough to justify taking permanent memory space IMO.

**suggestion:** Remove this declaration, and inline the string in the constructor directly.

Reply:
That's a good question, it has been directly inspired from what @paul-vd did on the OrderPlaced event : https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/core/events/domain/events/OrderPlacedEvent.ts?ref_type=heads#L24 maybe he can give us more details on this

Reply:
The reason was for future usage, so that we can expose it without creating an instance of order placed, for example in the cloud server events: https://gitlab.blackswift.cloud/front-commerce/cloud/front-commerce-scripts/-/blob/main/server-events-module/index.ts?ref_type=heads#L8,

Technically that should come from
```
OrderPlaced.event_type
```
Instead of defining an inline string ``"OrderPlaced"`` because if that changes between versions, we will now not have access to the updated event_type

Reply:
Restored since @paul-vd answer make sense to me